apis/externalsecrets/v1beta1/secretstore_conjur_types.go

```diff
@@ -0,0 +1,23 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+// // Configures an store to sync secrets using a Conjur backend.
+type ConjurProvider struct {
+	ServiceURL     *string `json:"serviceUrl,omitempty"`
+	ServiceUser    *string `json:"serviceUser,omitempty"`
+	ServiceAPIKey  *string `json:"serviceApiKey,omitempty"`
+	ServiceAccount *string `json:"serviceAccount,omitempty"`
```

Comment:
Conjur supports [different types of authentication](https://docs.conjur.org/Latest/en/Content/Developer/Conjur_API_Authenticate.htm) mechanisms, could we structure the CRD along the lines of the different means of authentication?

```yaml
# ...
conjur:
  auth:
    jwt:
      # server-hostname, service-id, account, host-id
    apiKey:
      # account, login, url
```

Reply:
While Conjur does support authn-jwt as an additional method of authentication, we would like more time to develop that piece out through our Community & Integrations team.  We do, however, have customers who are okay with using the secretStore as provided and would prefer to release this as-is with a new version including authn-jwt to come at a later time.

Would this be possible?

cc: @szh

Reply:
Regarding the SecretStore API structure: I think it works for now but once we want to add a new authentication method it gets quite messy because all the fields are on the same level. I'm not saying all authentication methods must be implemented right away, but we should design the structure in a way that allows us to easily add another one.

This is not only for the sake of maintainability, but also a UX concern. From a design perspective i see two areas:
1. connection/server related settings: A `url` to configure and TLS-related settings (note: other providers use `caBundle` and `caProvider` [see here for example](https://external-secrets.io/v0.8.1/api/clustersecretstore/)). Potentially also timeouts.
2. How to authenticate: (apiKey today, maybe more in the future)

Suggestion:
```yaml
# ...
    conjur:
      server:
        url: "https://example.tech"
        caBundle: "..."
        caProvider: {} # ...
      auth:
        apiKey:
          secretRef:
            name: "conjur-creds"
            key: "apikey"
``` 

The `serviceCertificate` would be renamed to `caBundle` with type `[]byte` which contains the CA cert. `caProvider` can be added at a later stage to pull the ca cert from a Secret/ConfigMap.

WDYT?